cmd/podmanV2/root.go

```diff
@@ -83,17 +134,62 @@ func syslogHook() {
-		registry.SetExitCode(0)
+func rootlessHook() {
+	if rootless.IsRootless() {
+		logrus.Error("rootless mode is currently not supported. Support will return ASAP.")
 	}
-	os.Exit(registry.GetExitCode())
+	// ce, err := registry.NewContainerEngine(rootCmd, []string{})
+	// if err != nil {
+	// 	logrus.WithError(err).Fatal("failed to obtain container engine")
+	// }
+	// ce.SetupRootLess(rootCmd)
+}
+
+func rootFlags(opts entities.PodmanConfig, flags *pflag.FlagSet) {
+	// V2 flags
+	flags.StringVarP(&opts.Uri, "remote", "r", "", "URL to access Podman service")
+	flags.StringSliceVar(&opts.Identities, "identity", []string{}, "path to SSH identity file")
+
+	// Override default --help information of `--version` global flag}
+	var dummyVersion bool
+	flags.BoolVar(&dummyVersion, "version", false, "Version of Podman")
+
+	cfg := opts.Config
+	flags.StringVar(&cfg.Engine.CgroupManager, "cgroup-manager", cfg.Engine.CgroupManager, opts.CGroupUsage)
+	flags.StringVar(&opts.CpuProfile, "cpu-profile", "", "Path for the cpu profiling results")
+	if err := flags.MarkHidden("cpu-profile"); err != nil {
+		logrus.WithError(err).Warn("unable to mark cpu-profile flag as hidden")
+	}
+	flags.StringSliceVar(&cfg.Engine.ConmonPath, "conmon", []string{}, "Path of the conmon binary")
+	flags.StringVar(&cfg.Engine.NetworkCmdPath, "network-cmd-path", cfg.Engine.NetworkCmdPath, "Path to the command for configuring the network")
+	flags.StringVar(&cfg.Network.NetworkConfigDir, "cni-config-dir", cfg.Network.NetworkConfigDir, "Path of the configuration directory for CNI networks")
+	flags.StringVar(&cfg.Containers.DefaultMountsFile, "default-mounts-file", cfg.Containers.DefaultMountsFile, "Path to default mounts file")
+	if err := flags.MarkHidden("default-mounts-file"); err != nil {
+		logrus.WithError(err).Warn("unable to mark default-mounts-file flag as hidden")
+	}
+	flags.StringVar(&cfg.Engine.EventsLogger, "events-backend", cfg.Engine.EventsLogger, `Events backend to use ("file"|"journald"|"none")`)
+	flags.StringSliceVar(&cfg.Engine.HooksDir, "hooks-dir", cfg.Engine.HooksDir, "Set the OCI hooks directory path (may be set multiple times)")
+	flags.IntVar(&opts.MaxWorks, "max-workers", 0, "The maximum number of workers for parallel operations")
+	if err := flags.MarkHidden("max-workers"); err != nil {
+		logrus.WithError(err).Warn("unable to mark max-workers flag as hidden")
+	}
+	flags.StringVar(&cfg.Engine.Namespace, "namespace", cfg.Engine.Namespace, "Set the libpod namespace, used to create separate views of the containers and pods on the system")
+	flags.StringVar(&cfg.Engine.StaticDir, "root", "", "Path to the root directory in which data, including images, is stored")
+	flags.StringVar(&opts.Runroot, "runroot", "", "Path to the 'run directory' where all state information is stored")
+	flags.StringSliceVar(&opts.Engine.RuntimePath, "runtime", []string{}, "Path to the OCI-compatible binary used to run containers, default is /usr/bin/runc")
```

Comment:
Should not be a string slice. Only one allowed.

Reply:
fixed